From fa1afb348504a15372f95ea4dba2072c0fc4d51f Mon Sep 17 00:00:00 2001
From: Jan-Benedikt Jagusch <jan.jagusch@quantco.com>
Date: Wed, 1 Sep 2021 11:00:22 +0300
Subject: [PATCH] Use more system libraries

---
 cmake/CMakeLists.txt | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 35822e2..d98cf08 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -761,7 +761,13 @@ get_filename_component(ORTTRAINING_ROOT "${ORTTRAINING_ROOT}" ABSOLUTE)
 get_filename_component(REPO_ROOT "${REPO_ROOT}" ABSOLUTE)
 set(ONNXRUNTIME_INCLUDE_DIR ${REPO_ROOT}/include/onnxruntime)
 
-add_subdirectory(external/date EXCLUDE_FROM_ALL)
+if(onnxruntime_PREFER_SYSTEM_LIB)
+  find_package(date)
+  add_library(date_interface ALIAS date::date)
+endif()
+if(NOT TARGET date::date)
+  add_subdirectory(external/date EXCLUDE_FROM_ALL)
+endif()
 
 set(SAFEINT_INCLUDE_DIR ${REPO_ROOT}/cmake/external/SafeInt)
 add_library(safeint_interface INTERFACE)
@@ -772,7 +778,12 @@ if(onnxruntime_DISABLE_EXCEPTIONS)
   add_compile_definitions(optional_CONFIG_NO_EXCEPTIONS=1)
 endif()
 
-add_subdirectory(external/mp11 EXCLUDE_FROM_ALL)
+if(onnxruntime_PREFER_SYSTEM_LIB)
+  find_package(boost_mp11)
+endif()
+if(NOT TARGET Boost::mp11)
+  add_subdirectory(external/mp11 EXCLUDE_FROM_ALL)
+endif()
 
 set(JSON_BuildTests OFF CACHE INTERNAL "")
 set(JSON_Install OFF CACHE INTERNAL "")
@@ -1269,7 +1280,13 @@ set(FLATBUFFERS_INSTALL OFF CACHE BOOL "FLATBUFFERS_INSTALL" FORCE)
 set(FLATBUFFERS_BUILD_FLATHASH OFF CACHE BOOL "FLATBUFFERS_BUILD_FLATHASH" FORCE)
 set(FLATBUFFERS_BUILD_FLATLIB ON CACHE BOOL "FLATBUFFERS_BUILD_FLATLIB" FORCE)
 set_msvc_c_cpp_compiler_warning_level(4)
-add_subdirectory(external/flatbuffers EXCLUDE_FROM_ALL)
+if(onnxruntime_PREFER_SYSTEM_LIB)
+  find_package(Flatbuffers)
+  add_library(flatbuffers ALIAS flatbuffers::flatbuffers)
+endif()
+if(NOT TARGET flatbuffers::flatbuffers)
+  add_subdirectory(external/flatbuffers EXCLUDE_FROM_ALL)
+endif()
 set_msvc_c_cpp_compiler_warning_level(3)
 list(APPEND onnxruntime_EXTERNAL_DEPENDENCIES flatbuffers)
 list(APPEND onnxruntime_EXTERNAL_LIBRARIES flatbuffers)
-- 
2.30.1
