diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index dbf0aca92..5d0d2db34 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -735,12 +735,7 @@ if (NOT WIN32)
   if (onnxruntime_PREFER_SYSTEM_LIB)
     find_package(nsync)
   endif()
-  if (TARGET nsync_cpp)  # linking error with nsync_FOUND (why?)
-    message("Use nsync from preinstalled system lib")
-  else()
-    message("Use nsync from submodule")
-    add_subdirectory(${PROJECT_SOURCE_DIR}/external/nsync EXCLUDE_FROM_ALL)
-  endif()
+  message("Use nsync from preinstalled system lib")
 endif()
 # External dependencies
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/external)
