diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index 3df351711a..968f4639f8 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -867,6 +867,7 @@ def update_submodules(source_dir):
 
 
 def install_python_deps(numpy_version=""):
+    return
     dep_packages = ["setuptools", "wheel", "pytest"]
     dep_packages.append(f"numpy=={numpy_version}" if numpy_version else "numpy>=1.16.6")
     dep_packages.append("sympy>=1.10")
