diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index 587d035541..0f8a9b6da2 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -1148,7 +1148,6 @@ def generate_build_tree(
                     f"Float 8 types require CUDA>=11.8. They must be disabled on CUDA=={args.cuda_version}. "
                     f"Add '--disable_types float8' to your command line. See option disable_types."
                 )
-        cmake_args.append(f"-DCMAKE_CUDA_COMPILER={cuda_home}/bin/nvcc")
     if args.use_rocm:
         cmake_args.append("-Donnxruntime_ROCM_HOME=" + rocm_home)
         cmake_args.append("-Donnxruntime_ROCM_VERSION=" + args.rocm_version)
