diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index eec7c5e33e..8d34d516c6 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -1441,7 +1441,6 @@ def generate_build_tree(
     if args.use_cuda:
         nvcc_threads = number_of_nvcc_threads(args)
         cmake_args.append("-Donnxruntime_NVCC_THREADS=" + str(nvcc_threads))
-        cmake_args.append(f"-DCMAKE_CUDA_COMPILER={cuda_home}/bin/nvcc")
         add_default_definition(cmake_extra_defines, "onnxruntime_USE_CUDA", "ON")
         if args.cuda_version:
             add_default_definition(cmake_extra_defines, "onnxruntime_CUDA_VERSION", args.cuda_version)
