From dd4946cb2d703f16f88764fca3513141bef0831e Mon Sep 17 00:00:00 2001
From: Christian Bourjau <christian.bourjau@quantco.com>
Date: Mon, 9 Sep 2024 18:21:45 +0200
Subject: [PATCH 1/4] Don't call pip on windows

---
 tools/ci_build/build.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index 1d2f889ff1..cca86d8469 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -2523,12 +2523,6 @@ def main():
             log.info("Activating emsdk...")
             run_subprocess([emsdk_file, "activate", emsdk_version], cwd=emsdk_dir)
 
-        if args.enable_pybind and is_windows():
-            run_subprocess(
-                [sys.executable, "-m", "pip", "install", "-r", "requirements/pybind/requirements.txt"],
-                cwd=SCRIPT_DIR,
-            )
-
         if args.use_rocm and args.rocm_version is None:
             args.rocm_version = ""
 
