From 4c3881927be996ad415e770219185d824342b0bb Mon Sep 17 00:00:00 2001
From: Christian Bourjau <christian.bourjau@quantco.com>
Date: Tue, 21 May 2024 12:45:50 +0200
Subject: [PATCH 3/4] stay on C++17 on osx

---
 cmake/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index baf21745e0..ce5c6da6cb 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -34,7 +34,7 @@ include(GNUInstallDirs) # onnxruntime_providers_* require CMAKE_INSTALL_* variab
 
 # TODO: update this once all system adapt c++20
 if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 17)
 else()
 set(CMAKE_CXX_STANDARD 17)
 endif()
