From 97f36e2a08f4036d0701dd78f1ecbf7d834ece36 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 25 May 2024 09:15:37 +0200
Subject: [PATCH 4/7] do not pass msvc style definitions to nvcc

---
 tools/ci_build/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index de72cf3b23..2246c20281 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -1142,7 +1142,7 @@ def generate_build_tree(
                     cuda_compile_flags_str = ""
                     for compile_flag in cflags:
                         if compile_flag.startswith("/D"):
-                            cudaflags.append(compile_flag)
+                            cudaflags.append(compile_flag.replace("/D","-D"))
                         else:
                             cuda_compile_flags_str = cuda_compile_flags_str + " " + compile_flag
                     if len(cuda_compile_flags_str) != 0:
